pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -52,26 +63,53 @@ func GetBaseDomain() (string, error) {
 	}
 	sort.Strings(publicZones)
 	if len(publicZones) == 0 {
-		return "", errors.New("no public Route 53 hosted zones found")
+		return nil, errors.New("no public Route 53 hosted zones found")
 	}
 
-	var domain string
 	if err := survey.AskOne(&survey.Select{
 		Message: "Base Domain",
 		Help:    "The base domain of the cluster. All DNS records will be sub-domains of this base and will also include the cluster name.\n\nIf you don't see you intended base-domain listed, create a new public Route53 hosted zone and rerun the installer.",
 		Options: publicZones,
-	}, &domain, func(ans interface{}) error {
+	}, &publicZoneName, func(ans interface{}) error {
 		choice := ans.(string)
 		i := sort.SearchStrings(publicZones, choice)
 		if i == len(publicZones) || publicZones[i] != choice {
 			return errors.Errorf("invalid base domain %q", choice)
 		}
 		return nil
 	}); err != nil {
-		return "", errors.Wrap(err, "failed UserInput for base domain")
+		return nil, errors.Wrap(err, "failed UserInput for base domain")
+	}
+
+	zoneList, ok := publicZoneMap[publicZoneName]
+	if !ok {
+		return nil, errors.Errorf("invalid base domain %q", publicZoneName)
+	}
+
+	zoneListLen := len(zoneList)
+	if zoneListLen == 0 {
+		return nil, errors.Wrap(err, "zero length zone list")
```

Comment:
It would be better if the error message explained the cause/suggested a fix. If I understand correctly, something like `"there must be at least one public zone for %s", publicZoneName`

Reply:
Fixed!